examples/vault-consul-ami/auth/sign-request.py

```diff
@@ -37,15 +39,17 @@ def generate_vault_request(awsIamServerId):
     request_dict['headers']['X-Vault-AWS-IAM-Server-ID'] = awsIamServerId
 
     request = endpoint.create_request(request_dict, operation_model)
+    request_headers_go_style = {k: [decode_if_bytes(v)] for k, v in request.headers.items()}
 
     return {
-        'iam_http_request_method': request.method,
-        'iam_request_url':         base64.b64encode(request.url),
-        'iam_request_body':        base64.b64encode(request.body),
-        'iam_request_headers':     base64.b64encode(json.dumps(headers_to_go_style(dict(request.headers)))), # It's a CaseInsensitiveDict, which is not JSON-serializable
+        'iam_http_request_method': decode_if_bytes(request.method),
+        'iam_request_url':         base64.b64encode(request.url.encode()).decode(),
```

Comment:
Why are we base 64 encoding and then immediately decoding?

Reply:
`decode` is different from `b64decode`. `decode` converts a python byte string (`b'hello world'`) to a python string (`'hello world'` - note the lack of `b`) (it means "decode utf-8"). Otherwise, when you do `json.dumps`, you get `{"key": "b'hello world'"}`.

Reply:
Roger